mikrotik/resource_interface_wireguard.go

```diff
@@ -0,0 +1,242 @@
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/int64default"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	tftypes "github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type interfaceWireguard struct {
+	client *client.Mikrotik
+}
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource                = &interfaceWireguard{}
+	_ resource.ResourceWithConfigure   = &interfaceWireguard{}
+	_ resource.ResourceWithImportState = &interfaceWireguard{}
+)
+
+// NewInterfaceWireguardResource is a helper function to simplify the provider implementation.
+func NewInterfaceWireguardResource() resource.Resource {
+	return &interfaceWireguard{}
+
+}
+
+func (i *interfaceWireguard) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	i.client = req.ProviderData.(*client.Mikrotik)
+}
+
+// Metadata returns the resource type name.
+func (i *interfaceWireguard) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_interface_wireguard"
+}
+
+// Schema defines the schema for the resource.
+func (i *interfaceWireguard) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Creates a Mikrotik interface wireguard.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				Description: "Identifier of this resource assigned by RouterOS",
+			},
+			"name": schema.StringAttribute{
+				Required:    true,
+				Description: "Name of the interface wireguard.",
+			},
+			"comment": schema.StringAttribute{
+				Optional:    true,
+				Description: "Comment associated with interface wireguard.",
+			},
+			"disabled": schema.BoolAttribute{
+				Optional:    true,
+				Computed:    true,
+				Default:     booldefault.StaticBool(false),
+				Description: "Boolean for whether or not the interface wireguard is disabled.",
+			},
+			"listen_port": schema.Int64Attribute{
+				Optional:    true,
+				Computed:    true,
+				Default:     int64default.StaticInt64(13231),
+				Description: "Port for WireGuard service to listen on for incoming sessions.",
+			},
+			"mtu": schema.Int64Attribute{
+				Optional:    true,
+				Computed:    true,
+				Default:     int64default.StaticInt64(1420),
+				Description: "Layer3 Maximum transmission unit.",
```

Comment:
This is a minor fix, but we could add validation to this field. The `Validators` field is compatible with the terraform-plugin-framework-validator package and supports the [Between function](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework-validators@v0.10.0/int64validator#Between). This value should be between 0 and 65536 according to the mikrotik docs.

<img width="992" alt="Screen Shot 2023-05-10 at 11 20 12 PM" src="https://github.com/ddelnano/terraform-provider-mikrotik/assets/5855593/c6d6b9b9-183d-40ae-b6c5-d601a1284daf">




I know this is your first change, so feel free to skip this. However, I think it's very useful to know the flexibility with the validation hooks.

Reply:
Added! Still need to create tests to make sure it is validating properly

Reply:
Sounds good. I don't think an acceptance test is necessary for validating this logic. There is always a tradeoff for if additional test coverage is worth it. Since the validator has its [own unit tests](https://github.com/hashicorp/terraform-plugin-framework-validators/blob/5fd6bf61245611b1aa74c0b5a421c4d630efe29a/int64validator/between_test.go#L14), I think it's ok to skip it as long as the happy path (an int value that passes validation) works. The happy path would be covered by our existing acceptance tests.